hal/tests/TESTS/CMakeLists.txt

```diff
@@ -0,0 +1,38 @@
+# Copyright (c) 2021 ARM Limited. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+add_subdirectory(mbed_hal/common_tickers)
+add_subdirectory(mbed_hal/common_tickers_freq)
+add_subdirectory(mbed_hal/crc)
+add_subdirectory(mbed_hal/critical_section)
+add_subdirectory(mbed_hal/flash/functional_tests)
+add_subdirectory(mbed_hal/lp_ticker)
+add_subdirectory(mbed_hal/minimum_requirements)
+add_subdirectory(mbed_hal/mpu)
+add_subdirectory(mbed_hal/pinmap)
+add_subdirectory(mbed_hal/reset_reason)
+add_subdirectory(mbed_hal/rtc)
+add_subdirectory(mbed_hal/rtc_reset)
+add_subdirectory(mbed_hal/rtc_time)
+add_subdirectory(mbed_hal/rtc_time_conv)
+add_subdirectory(mbed_hal/sleep)
+add_subdirectory(mbed_hal/sleep_manager)
+add_subdirectory(mbed_hal/sleep_manager_racecondition)
+add_subdirectory(mbed_hal/ticker)
+add_subdirectory(mbed_hal/trng)
+add_subdirectory(mbed_hal/us_ticker)
+add_subdirectory(mbed_hal/watchdog)
+add_subdirectory(mbed_hal/watchdog_reset)
+add_subdirectory(mbed_hal/watchdog_timing)
+add_subdirectory(mbed_hal/qspi)
+add_subdirectory(mbed_hal/ospi)
+add_subdirectory(mbed_hal_fpga_ci_test_shield/analogin)
+add_subdirectory(mbed_hal_fpga_ci_test_shield/gpio)
+add_subdirectory(mbed_hal_fpga_ci_test_shield/gpio_irq)
+add_subdirectory(mbed_hal_fpga_ci_test_shield/i2c)
+add_subdirectory(mbed_hal_fpga_ci_test_shield/pwm)
+add_subdirectory(mbed_hal_fpga_ci_test_shield/spi)
+add_subdirectory(mbed_hal_fpga_ci_test_shield/uart)
+add_subdirectory(mbed_timing_fpga_ci_test_shield/watchdog)
+add_subdirectory(pin_names/arduino_uno)
+add_subdirectory(pin_names/generic)
```

Comment:
That's a lot of directories! I don't really understand what this commit is for after reading the commit message. Could we add a bit more context as to why we need this? Should it be in a separate commit from "Update hal CMake to include greentea test dir"?